test_redir.php

```diff
@@ -6,7 +6,11 @@
 }
 
 // Redirect to full self URL.
-$testurl = 'http://'.$_SERVER['SERVER_NAME'].':'.$_SERVER['SERVER_PORT'].$_SERVER['SCRIPT_NAME'];
+$scheme = 'http';
+if (!empty($_SERVER['HTTPS']) && strcasecmp($_SERVER['HTTPS'], 'off') !== 0) {
+    $scheme = 'https';
+}
+$testurl = $scheme.'://'.$_SERVER['SERVER_NAME'].':'.$_SERVER['SERVER_PORT'].$_SERVER['SCRIPT_NAME'];
```

Comment:
Since this line is changing anyway, maybe add spaces?
```suggestion
$testurl = $scheme . '://' . $_SERVER['SERVER_NAME'] . ':' . $_SERVER['SERVER_PORT'] . $_SERVER['SCRIPT_NAME'];
```